inst/string.m

```diff
@@ -1276,3 +1284,31 @@ function disp (this)
     endif
   endfor
 endfunction
+
+## Test string constructor
+%!test
+%! str = string (["a";"b";"c"]);
+%! assert (cell (str), {"a";"b";"c"})
```

Comment:
Let's use `cellstr` instead of `cell` in the test, since that's the one that's part of the standard M-code `string` interface, and we probably want to encourage people to use. The `cell` method is a compatibility hack intended mostly for implicit use in assignment or generic programming contexts.